test/test_retry.py

```diff
@@ -131,6 +131,18 @@ def test_zero_backoff(self):
         retry = retry.increment()
         self.assertEqual(retry.get_backoff_time(), 0)
 
+    def test_backoff_reset_after_redirect(self):
+        retry = Retry(total=100, redirect=5, backoff_factor=0.2)
+        retry = retry.increment()
+        retry = retry.increment()
+        self.assertEqual(retry.get_backoff_time(), 0.4)
+        redirect_response = HTTPResponse(status=302, headers={'location': 'test'})
+        retry = retry.increment(response=redirect_response)
+        self.assertEqual(retry.get_backoff_time(), 0)
+        retry = retry.increment()
+        retry = retry.increment()
+        self.assertEqual(retry.get_backoff_time(), 0.4)
+
```

Comment:
This test is not checking that we increment exponentially. The test check that we reset after a redirect. I actually think that I should remove one of the  ``retry.increment()`` at each stage as it is not contributing to this test purpose.